pkg/virt-controller/watch/vm.go

```diff
@@ -3056,9 +3056,11 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 
 		if syncErr == nil {
 			if !equality.Semantic.DeepEqual(vm, vmCopy) {
-				vm, err = c.clientset.VirtualMachine(vmCopy.Namespace).Update(context.Background(), vmCopy)
+				updatedVm, err := c.clientset.VirtualMachine(vmCopy.Namespace).Update(context.Background(), vmCopy)
```

Comment:
This whole `sync` method is a mess. The fact we need this complexity here is a sign of its problem.

It is very hard to even understand on what changes this`Update` works on. The only hint is the error message.

Nevertheless, the small fix seems correct.
You could embed it in the condition (`if`) but it is nit.